.cbmc-batch/jobs/aws_cryptosdk_cmm_generate_enc_materials/aws_cryptosdk_cmm_generate_enc_materials_harness.c

```diff
@@ -0,0 +1,135 @@
+#include <proof_helpers/proof_allocators.h>
+#include <proof_helpers/utils.h>
+
+// Stub this until https://github.com/diffblue/cbmc/issues/5344 is fixed
+// Original function is here:
+// https://github.com/aws/aws-encryption-sdk-c/blob/master/source/edk.c#L44
+void aws_cryptosdk_edk_list_clean_up(struct aws_array_list *encrypted_data_keys) {
+    assert(aws_cryptosdk_edk_list_is_valid(encrypted_data_keys));
+    aws_array_list_clean_up(encrypted_data_keys);
+}
+
+/**
+ * Receives encryption request from user and attempts to generate encryption materials,
+ * including an encrypted data key and a list of EDKs for doing encryption.
+ *
+ * On success returns AWS_OP_SUCCESS and allocates encryption materials object at address
+ * pointed to by output.
+ *
+ * On failure returns AWS_OP_ERR, sets address pointed to by output to NULL, and sets
+ * internal AWS error code.
+ */
+int generate_enc_materials(
+    struct aws_cryptosdk_cmm *cmm,
+    struct aws_cryptosdk_enc_materials **output,
+    struct aws_cryptosdk_enc_request *request) {
+    // assert(aws_cryptosdk_cmm_base_is_valid(cmm));
+    // assert(__CPROVER_w_ok(output, sizeof(*output)));
+    // assert(aws_cryptosdk_enc_request_is_valid(request));
+
+    struct aws_cryptosdk_enc_materials *materials = can_fail_malloc(sizeof(*materials));
+    if (materials == NULL) {
+        *output = NULL;
+        return AWS_OP_ERR;
+    }
+
+    // Set up the allocator
+    materials->alloc = request->alloc;
+    __CPROVER_assume(aws_allocator_is_valid(materials->alloc));
+
+    // Set up the signctx
+    materials->signctx = can_fail_malloc(sizeof(*materials->signctx));
+    if (materials->signctx) {
+        ensure_sig_ctx_has_allocated_members(materials->signctx);
+        __CPROVER_assume(aws_cryptosdk_sig_ctx_is_valid_cbmc(materials->signctx));
+    }
+
+    // Set up the unencrypted_data_key
+    __CPROVER_assume(aws_byte_buf_is_bounded(&materials->unencrypted_data_key, MAX_NUM_ITEMS));
+    ensure_byte_buf_has_allocated_buffer_member(&materials->unencrypted_data_key);
+    __CPROVER_assume(aws_byte_buf_is_valid(&materials->unencrypted_data_key));
+
+    // Set up the edk_list
+    // edk_list Precondition: We have a valid list */
+    __CPROVER_assume(aws_cryptosdk_edk_list_is_bounded(&materials->encrypted_data_keys, MAX_NUM_ITEMS));
+    ensure_cryptosdk_edk_list_has_allocated_list(&materials->encrypted_data_keys);
+    __CPROVER_assume(aws_cryptosdk_edk_list_is_valid(&materials->encrypted_data_keys));
+
+    // Stub until https://github.com/diffblue/cbmc/issues/5344 is fixed
+    // edk_list Precondition: The list has valid list elements
+    /*
+      __CPROVER_assume(aws_cryptosdk_edk_list_elements_are_bounded(&materials->encrypted_data_keys, MAX_STRING_LEN));
+      ensure_cryptosdk_edk_list_has_allocated_list_elements(&materials->encrypted_data_keys);
+      __CPROVER_assume(aws_cryptosdk_edk_list_elements_are_valid(&materials->encrypted_data_keys));
+    */
```

Comment:
We should have an issue here to update these proofs once #5344 is fixed.

Reply:
https://github.com/aws/aws-encryption-sdk-c/issues/529